pkg/schemacompat/schemacompat.go

```diff
@@ -343,46 +347,49 @@ func lcdForObject(fldPath *field.Path, existing, new *schema.Structural, lcd *sc
+				multierr.AppendInto(&err, lcdForStructural(fldPath.Child("properties").Key(key), &existingPropertySchema, new.AdditionalProperties.Structural, &lcdPropertySchema, narrowExisting))
 				lcd.Properties[key] = lcdPropertySchema
 			}
 		} else if new.AdditionalProperties.Bool {
 			// that allows named properties only.
 			// => Keep the existing schemas as the lcd.
 		} else {
-			errorList = append(errorList, field.Invalid(fldPath.Child("properties"), sets.StringKeySet(existing.Properties).List(), "properties value has been completely cleared in an incompatible way"))
+			multierr.AppendInto(&err, field.Invalid(fldPath.Child("properties"), sets.StringKeySet(existing.Properties).List(), "properties value has been completely cleared in an incompatible way"))
 		}
 	} else if existing.AdditionalProperties != nil {
 		if existing.AdditionalProperties.Structural != nil {
 			if new.AdditionalProperties.Structural != nil {
-				errorList = append(errorList, lcdForStructural(fldPath.Child("additionalProperties"), existing.AdditionalProperties.Structural, new.AdditionalProperties.Structural, lcd.AdditionalProperties.Structural, narrowExisting)...)
+				multierr.AppendInto(&err, lcdForStructural(fldPath.Child("additionalProperties"), existing.AdditionalProperties.Structural, new.AdditionalProperties.Structural, lcd.AdditionalProperties.Structural, narrowExisting))
 			} else if existing.AdditionalProperties != nil && new.AdditionalProperties.Bool {
 				// new schema allows any properties of any schema here => it is a superset of the existing schema
 				// that allows any properties of a given schema.
 				// => Keep the existing schemas as the lcd.
 			} else {
-				errorList = append(errorList, field.Invalid(fldPath.Child("additionalProperties"), new.AdditionalProperties.Bool, "additionalProperties value has been changed in an incompatible way"))
+				multierr.AppendInto(&err, field.Invalid(fldPath.Child("additionalProperties"), new.AdditionalProperties.Bool, "additionalProperties value has been changed in an incompatible way"))
 			}
 		} else if existing.AdditionalProperties.Bool {
 			if !new.AdditionalProperties.Bool {
 				if !narrowExisting {
-					errorList = append(errorList, field.Invalid(fldPath.Child("additionalProperties"), new.AdditionalProperties.Bool, "additionalProperties value has been changed in an incompatible way"))
+					multierr.AppendInto(&err, field.Invalid(fldPath.Child("additionalProperties"), new.AdditionalProperties.Bool, "additionalProperties value has been changed in an incompatible way"))
 				}
 				lcd.AdditionalProperties.Bool = false
 				lcd.AdditionalProperties.Structural = new.AdditionalProperties.Structural
 			}
 		}
-	} else {
-		// Existing schema doesn't allow anything => new will always be a superset of existing
 	}
 
-	return
+	multierr.AppendInto(&err, lcdForObjectValidation(fldPath, existing.ValueValidation, new.ValueValidation, lcd.ValueValidation, narrowExisting))
```

Comment:
This was found by linting, because `lcdForObjectValidation` was never called previously. Let me know if this matches your expectation for when/how this should be called.